tests_e2e/orchestrator/lib/agent_test_loader.py

```diff
@@ -128,18 +128,26 @@ def _validate(self):
         """
         Performs some basic validations on the data loaded from the YAML description files
         """
+        def _parse_image(image: str) -> str:
+            """
+            Parses a reference to an image or image set and returns the name of the image or image set
+            """
+            match = AgentTestLoader.RANDOM_IMAGES_RE.match(image)
+            if match is not None:
+                return match.group('image_set')
+            return image
+
         for suite in self.test_suites:
             # Validate that the images the suite must run on are in images.yml
             for image in suite.images:
-                match = AgentTestLoader.RANDOM_IMAGES_RE.match(image)
-                if match is not None:
-                    image = match.group('image_set')
+                image = _parse_image(image)
                 if image not in self.images:
                     raise Exception(f"Invalid image reference in test suite {suite.name}: Can't find {image} in images.yml")
 
             # If the suite specifies a location, validate that the images it uses are available in that location
             if suite.location != '':
                 for suite_image in suite.images:
+                    suite_image = _parse_image(suite_image)
                     for image in self.images[suite_image]:
```

Comment:
suite_image can be "random(endorsed)", so we have to parse here as well to get image_set or image name